src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ReviewTimeExtensionsHandler.java

```diff
@@ -79,8 +148,32 @@ private String getTimeExtensionDecisionReason(AsylumCase asylumCase) {
         return read.orElseThrow(() -> new IllegalArgumentException("Cannot handle " + Event.REVIEW_TIME_EXTENSION + " without a decision reason"));
     }
 
-    private Stream<IdValue<TimeExtension>> getTimeExtensions(AsylumCase asylumCase) {
+    private String getTimeExtensionDueDate(AsylumCase asylumCase) {
+        Optional<String> read = asylumCase.read(REVIEW_TIME_EXTENSION_DUE_DATE);
+        return read.orElseThrow(() -> new IllegalArgumentException("Cannot handle " + Event.REVIEW_TIME_EXTENSION + " without a decision reason"));
```

Comment:
Not sure this is correct as you are getting a due date. Should we really throw an exception is this something that can't really happen or should it be an an error added to the response which is then shown to the user.